client/src/main/java/org/asynchttpclient/netty/channel/NettyConnectListener.java

```diff
@@ -94,12 +106,7 @@ public void onSuccess(Channel channel, InetSocketAddress remoteAddress) {
             Object partitionKeyLock = future.takePartitionKeyLock();
 
             if (partitionKeyLock != null) {
-                channel.closeFuture().addListener(new GenericFutureListener<Future<? super Void>>() {
-                    @Override
-                    public void operationComplete(Future<? super Void> future) throws Exception {
-                        connectionSemaphore.releaseChannelLock(partitionKeyLock);
-                    }
-                });
+                channel.closeFuture().addListener(future -> connectionSemaphore.releaseChannelLock(partitionKeyLock));
```

Comment:
Agree, but not related to this feature. Please revert (I'll push it right away).